src/libs/ascent/python/ascent_module/ascent_mpi_python.cpp

```diff
@@ -465,6 +488,72 @@ PyAscent_MPI_about()
     return (PyObject*)py_node_res;
 }
 
+//---------------------------------------------------------------------------//
+// ascent::execute_callback
+//---------------------------------------------------------------------------//
+static PyObject *
+PyAscent_MPI_execute_callback(PyObject *self,
+                              PyObject *args)
+{
+    char *callback_name;
+    PyObject *py_params = NULL;
+    PyObject *py_output = NULL;
+
+    if (!PyArg_ParseTuple(args,
+                          "sOO",
```

Comment:
we can use `s|00` to make all the args optional (maybe to support the bool callback case)